src/Microsoft.IdentityModel.Logging/TextWriterEventListener.cs

```diff
@@ -31,7 +32,7 @@ public TextWriterEventListener()
                 _streamWriter = new StreamWriter(fileStream);
                 _streamWriter.AutoFlush = true;
             }
-            catch (Exception ex)
+            catch (Exception ex) when (LogHelper.IsEnabled(EventLogLevel.Error))
```

Comment:
@stephentoub will this exception still be thrown even with logging not enabled?

Reply:
Yes. If logging is disabled, that when filter will make it behave as if the catch block didn't exist at all, and the original exception will propagate. It doesn't change the observed behavior and makes it cheaper when logging isn't enabled as we're then avoiding catching the exception only to immediately rethrow it.